### PR TITLE
BUG-104774 – extended exception message for rds deletion

### PR DESCRIPTION
**Pull request by gregito**
the exception when the rds is going to be deleted but it is used by an active cluster now returns two different message, depending on the amount of clusters which uses it.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java [hunk 1]
```diff
@@ -146,15 +146,20 @@ private void delete(RDSConfig rdsConfig) {
...
+                for (int i = 0; i < clustersWithProvidedRds.size() - 1; i++) {
+                    clusters.append(clustersWithProvidedRds.get(i).getName()).append(", ");
+                }
+                clusters.append(clustersWithProvidedRds.get(clustersWithProvidedRds.size() - 1).getName()).append(']');
+                throw new BadRequestException(String.format(
+                        "There are clusters associated with RDS config '%s'. Please remove these before deleting the RDS configuration. "
+                                + "The following clusters are using this RDS: %s", rdsConfig.getName(), clusters.toString()));
+            } else {
+                throw new BadRequestException(String.format("There is a cluster ['%s'] which uses RDS config '%s'. Please remove this "
+                        + "cluster before deleting the blueprint", clustersWithProvidedRds.get(0).getName(), rdsConfig.getName()));
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java]: blueprint?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java [hunk 2]
```diff
@@ -146,15 +146,20 @@ private void delete(RDSConfig rdsConfig) {
     private void checkRdsConfigNotAssociated(RDSConfig rdsConfig) {
         List<Cluster> clustersWithProvidedRds = new ArrayList<>(clusterRepository.findAllClustersByRDSConfig(rdsConfig.getId()));
         if (!clustersWithProvidedRds.isEmpty()) {
-            StringBuilder clusters = new StringBuilder();
-            clusters.append('[');
-            for (int i = 0; i < clustersWithProvidedRds.size() - 1; i++) {
-                clusters.append(clustersWithProvidedRds.get(i).getName()).append(", ");
+            if (clustersWithProvidedRds.size() > 1) {
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java]: !isempty()?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java [hunk 3]
```diff
@@ -146,15 +146,20 @@ private void delete(RDSConfig rdsConfig) {
...
+                StringBuilder clusters = new StringBuilder();
+                clusters.append('[');
+                for (int i = 0; i < clustersWithProvidedRds.size() - 1; i++) {
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java]: String.join ?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java (hunk 1 above)
Comment (gregito) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java]: my fault, going to change :)

File: core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java (hunk 2 above)
Comment (gregito) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java]: like in other cases, different message come back if the related clusters size is *more* than one

**Pull request merged**